### PR TITLE
[AzureAD] Don't pass resource when requesting a token

### PR DESCRIPTION
**Pull request by craigminihan**
* `AzureAdOAuthenticator` fails when calling into Azure AD when the `resource` field is present in the token body.
* Updated the example config.py to show how `username_claim` can be used when the `name` returned by AD isn't compliant with JH.
* Minor typo change

### CODE REVIEW

File: oauthenticator/azuread.py [hunk 1]
```diff
@@ -77,7 +76,7 @@ def _username_claim_default(self):
 
         headers = {
             'Content-Type':
-            'application/x-www-form-urlencoded; ; charset=UTF-8"'
```
Comment (craigminihan) [inline oauthenticator/azuread.py]: This looks malformed so updated

File: oauthenticator/azuread.py [hunk 2]
```diff
@@ -67,7 +67,6 @@ def _username_claim_default(self):
             client_secret=self.client_secret,
             grant_type='authorization_code',
             code=code,
-            resource=self.client_id,
```
Comment (craigminihan) [inline oauthenticator/azuread.py]: Azure responds with 400 if this field is present

File: oauthenticator/azuread.py [hunk 3]
```diff
@@ -19,13 +19,13 @@
 from .oauth2 import OAuthLoginHandler, OAuthenticator
 
 
-def azure_token_url_for(tentant):
```
Comment (craigminihan) [inline oauthenticator/azuread.py]: super minor spelling issue

File: examples/azuread/sample_jupyter_config.py [hunk 4]
```diff
@@ -9,3 +9,8 @@
 c.AzureAdOAuthenticator.oauth_callback_url = 'http://{your-domain}/hub/oauth_callback'
 c.AzureAdOAuthenticator.client_id = '{AAD-APP-CLIENT-ID}'
 c.AzureAdOAuthenticator.client_secret = '{AAD-APP-CLIENT-SECRET}'
+
+# if the user's name returned by Azure isn't acceptable to JupyterHub then we can use an alternate field,
+# uncomment the line below to use 'unique_name' rather than the default 'name'. Consult the Azure
+# documentation for other field names.
+#c.AzureAdOAuthenticator.username_claim = 'unique_name'
```
Comment (craigminihan) [inline examples/azuread/sample_jupyter_config.py]: For me the `name` returned by AD is bounced by JH so I had to use another value. This field was already present in the class, setting its value to `unique_name` solved my problem. I added this to the example code for others to more easily find than me.

See: https://github.com/jupyterhub/jupyterhub/blob/master/jupyterhub/auth.py#L185

**Comment by craigminihan** [comment]
I created a custom image based on `jupyterhub/k8s-hub:0.8.2` with the `azurerd.py` from this PR. I have been able to log into JH via the OAuth redirect to a corporate sync'd AD instance in Azure.

The Dockerfile is as follows:
```
FROM jupyterhub/k8s-hub:0.8.2

USER root
RUN apt-get update && \
    apt-get install -y wget

USER jovyan

RUN pip3 install --user --no-deps --no-compile oauthenticator==0.8.0 && \
    wget https://raw.githubusercontent.com/craigminihan/oauthenticator/remove_resource/oauthenticator/azuread.py -O ~/.local/lib/python3.6/site-packages/oauthenticator/azuread.py
```
The updated `azurerd` is loaded by JH from `.local` here.

**Comment by minrk** [comment]
Thanks!

**Pull request merged**